internal/pkg/manifest/svc.go

```diff
@@ -39,7 +49,112 @@ type Service struct {
 
 // ServiceImage represents the service's container image.
 type ServiceImage struct {
-	Build *string `yaml:"build"` // Path to the Dockerfile.
+	Build BuildArgsOrString `yaml:"build"` // Path to the Dockerfile.
+}
+
+// BuildConfig populates a docker.BuildArguments struct from the fields available in the manifest.
+// Prefer the following hierarchy:
+// 1. Specific dockerfile, specific context
+// 2. Specific dockerfile, context = dockerfile dir
+// 3. "Dockerfile" located in context dir
+// 4. "Dockerfile" located in ws root.
+func (s *ServiceImage) BuildConfig(rootDirectory string) *docker.BuildArguments {
+	df := s.dockerfile()
+	ctx := s.context()
+	if df != "" && ctx != "" {
+		df = filepath.Join(rootDirectory, df)
+		ctx = filepath.Join(rootDirectory, ctx)
+	} else if df != "" && ctx == "" {
+		df = filepath.Join(rootDirectory, df)
+		ctx = filepath.Dir(df)
+	} else if df == "" && ctx != "" {
+		ctx = filepath.Join(rootDirectory, ctx)
+		df = filepath.Join(ctx, dockerfileDefaultName)
+	} else {
+		ctx = rootDirectory
+		df = filepath.Join(rootDirectory, dockerfileDefaultName)
+	}
+	return &docker.BuildArguments{
+		Dockerfile: df,
+		Context:    ctx,
+		Args:       s.args(),
+	}
+}
+
+// dockerfile returns the path to the service's Dockerfile. If no dockerfile is specified,
+// returns "".
+func (s *ServiceImage) dockerfile() string {
+	// Prefer to use the "Dockerfile" string in BuildArgs. Otherwise,
+	// "BuildString". If no dockerfile specified, return "".
+	if s.Build.BuildArgs.Dockerfile != nil {
+		return aws.StringValue(s.Build.BuildArgs.Dockerfile)
+	}
+
+	var dfPath string
+	if s.Build.BuildString != nil {
+		dfPath = aws.StringValue(s.Build.BuildString)
+	} else {
+		dfPath = ""
+	}
+
+	return dfPath
+}
+
+// context returns the build context directory if it exists, otherwise an empty string.
+func (s *ServiceImage) context() string {
+	return aws.StringValue(s.Build.BuildArgs.Context)
+}
+
+// args returns the args section, if it exists, to override args in the dockerfile.
+// Otherwise it returns an empty map.
+func (s *ServiceImage) args() map[string]string {
+	return s.Build.BuildArgs.Args
+}
+
+// BuildArgsOrString is a custom type which supports unmarshaling yaml which
+// can either be of type string or type DockerBuildArgs.
+type BuildArgsOrString struct {
+	BuildString *string
+	BuildArgs   DockerBuildArgs
+}
+
+// UnmarshalYAML overrides the default YAML unmarshaling logic for the BuildArgsOrString
+// struct, allowing it to perform more complex unmarshaling behavior.
+// This method implements the yaml.Unmarshaler (v2) interface.
+func (b *BuildArgsOrString) UnmarshalYAML(unmarshal func(interface{}) error) error {
+	if err := unmarshal(&b.BuildArgs); err != nil {
+		switch err.(type) {
+		case *yaml.TypeError:
+			break
+		default:
+			return err
+		}
+	}
+
+	if !b.BuildArgs.isEmpty() {
+		return nil
+	}
```

Comment:
nit: can we add a comment here like:
```suggestion
	if !b.BuildArgs.isEmpty() {
		// Unmarshaled successfully to b.BuildArgs, return.
		return nil
	}
```